pkg/kritis/gcbsigner/signer.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gcbsigner
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/grafeas/kritis/pkg/kritis/apis/kritis/v1beta1"
+	"github.com/grafeas/kritis/pkg/kritis/crd/authority"
+	"github.com/grafeas/kritis/pkg/kritis/crd/buildpolicy"
+	"github.com/grafeas/kritis/pkg/kritis/metadata"
+	"github.com/grafeas/kritis/pkg/kritis/secrets"
+	containeranalysispb "google.golang.org/genproto/googleapis/devtools/containeranalysis/v1alpha1"
+)
+
+type Signer struct {
+	config *Config
+	client metadata.Fetcher
+}
+
+type Config struct {
+	Secret   secrets.Fetcher
+	Validate buildpolicy.ValidateFunc
+}
+
+func New(client metadata.Fetcher, c *Config) Signer {
+	return Signer{
+		client: client,
+		config: c,
+	}
+}
+
+type ImageBuildInfo struct {
```

Comment:
Omit words that are obvious given a variable's type declaration & omit words that are clear from the surrounding context. 

"Info" isn't terribly helpful for naming. What do you think about "ImageBuild"?

Reply:
How about BuildProvenance?  It's a bit of a stretch as the info in there is a bit thin at the moment, but that is what this really represents.